mldsa/poly.c

```diff
@@ -299,27 +301,24 @@ __contract__(
 
 void poly_uniform(poly *a, const uint8_t seed[MLDSA_SEEDBYTES], uint16_t nonce)
 {
-  unsigned int i, ctr, off;
+  unsigned int ctr;
   unsigned int buflen = POLY_UNIFORM_NBLOCKS * STREAM128_BLOCKBYTES;
-  uint8_t buf[POLY_UNIFORM_NBLOCKS * STREAM128_BLOCKBYTES + 2];
+  uint8_t buf[POLY_UNIFORM_NBLOCKS * STREAM128_BLOCKBYTES];
   stream128_state state;
 
   stream128_init(&state, seed, nonce);
   stream128_squeezeblocks(buf, POLY_UNIFORM_NBLOCKS, &state);
 
-  ctr = rej_uniform(a->coeffs, MLDSA_N, buf, buflen);
-
+  ctr = rej_uniform(a->coeffs, MLDSA_N, 0, buf, buflen);
+  buflen = STREAM128_BLOCKBYTES;
```

Comment:
`rej_uniform` doesn't modify the value of `buflen` right? Then there should be no need to reset it each iteration of the loop, as[ upstream kyber did](https://github.com/pq-crystals/kyber/commit/ce492cd). I noticed it got moved out of the loop in [mlkem-native](https://github.com/pq-code-package/mlkem-native/blob/main/mlkem/sampling.c#L220).